zip-0317.rst

```diff
@@ -78,36 +79,61 @@ Requirements
-====================== ==============================================
-:math:`marginal\_fee`  zatoshis per logical action (as defined below)
-:math:`grace\_actions` logical actions
-====================== ==============================================
+===================================== ==============================================
+Parameter                             Units
+===================================== ==============================================
+:math:`marginal\_fee`                 zatoshis per logical action (as defined below)
+:math:`grace\_actions`                logical actions
+:math:`p2pkh\_standard\_input\_size`  bytes
+:math:`p2pkh\_standard\_output\_size` bytes
+===================================== ==============================================
 
 Wallets implementing this specification SHOULD use a conventional fee
 calculated in zatoshis per the following formula:
 
 .. math::
 
    \begin{array}{rcl}
-     logical\_actions  &=& \max(transparent\_inputs, transparent\_outputs) \;+ \\
-                       & & 2 \cdot sprout\_joinsplits \;+ \\
-                       & & \max(sapling\_inputs, sapling\_outputs) \;+ \\
-                       & & orchard\_actions \\
-     conventional\_fee &=& marginal\_fee \cdot \max(grace\_actions, logical\_actions)
+     logical\_actions  &=& \mathsf{max}\big(\mathsf{ceiling}\big(\frac{tx\_in\_total\_size}{p2pkh\_standard\_input\_size}\big),
+                                            \mathsf{ceiling}\big(\frac{tx\_out\_total\_size}{p2pkh\_standard\_output\_size}\big)\big) \;+ \\
+                       & & 2 \cdot nJoinSplits \;+ \\
+                       & & \mathsf{max}(nSpendsSapling, nOutputsSapling) \;+ \\
+                       & & nActionsOrchard \\
+     conventional\_fee &=& marginal\_fee \cdot \mathsf{max}(grace\_actions, logical\_actions)
    \end{array}
 
+The inputs to this formula are taken from transaction fields defined in the Zcash protocol
+specification [#protocol-txnencoding]:
+
+============================ ====== ===========================================
+Input                        Units  Description
+============================ ====== ===========================================
+:math:`tx\_in\_total\_size`  bytes  total size in bytes of the ``tx_in`` field
+:math:`tx\_out\_total\_size` bytes  total size in bytes of the ``tx_out`` field
+:math:`nJoinSplit`           number the number of Sprout JoinSplits
+:math:`nSpendsSapling`       number the number of Sapling spends
+:math:`nOutputsSapling`      number the number of Sapling outputs
+:math:`nActionsOrchard`      number the number of Orchard actions
+============================ ====== ===========================================
+
 The parameters are set to the following values:
 
 * :math:`marginal\_fee = 5000`;
-* :math:`grace\_actions = 2`.
+* :math:`grace\_actions = 2`;
+* :math:`p2pkh\_standard\_input\_size = TODO` bytes;
+* :math:`p2pkh\_standard\_output\_size = TODO` bytes.
```

Comment:
```suggestion
* :math:`p2pkh\_standard\_input\_size = 150` bytes;
* :math:`p2pkh\_standard\_output\_size = 34` bytes.
```

Reply:
For per-input sizes the values can range from 147 to 149 bytes due to ❤️‍🔥 just ❤️‍🔥 how ❤️‍🔥 lovely ❤️‍🔥 the DER encoding of a secp256k1 signature is. 150 bytes seems fine for this purpose.